ols/src/prompts/prompts.py

```diff
@@ -10,15 +10,20 @@
 # TODO: Fine-tune system prompt
 QUERY_SYSTEM_PROMPT = """You are an assistant for question-answering tasks \
 related to the openshift and kubernetes container orchestration platforms. \
-Use the previous chat history to interact and help the user.
+"""
+
+USE_PREVIOUS_HISTORY = "Use the previous chat history to interact and help the user.\n"
+USE_RETRIEVED_CONTEXT = """
 Use the following pieces of retrieved context to answer the question.
 
 {context}
 """
 
 CHAT_PROMPT = ChatPromptTemplate.from_messages(
```

Comment:
should this ever be used any more?  shouldn't we always have a dynamically generated prompt?

Reply:
It's used in `DocsSummarizer._get_rag_data`. Seems like chicken-egg problem there, as we want to generate prompt, but in order to do it, we need RAG context taken using (generated?) prompt. @asamal4 WDYT?

Reply:
so the point of https://github.com/openshift/lightspeed-service/blob/71725e35ca6ed7ef018000d68de30f9a1cba59aa/ols/src/query_helpers/docs_summarizer.py#L80 is to estimate what the overall prompt length is going to be, prior to including any rag content, so we can decide how much space we can dedicate to including rag content.

Note that because our priority of inclusion is:
1) rag content
2) conversation history

the interim computation assumes zero conversation history.

for purposes of the computation, we should still be calling the function to generate the dynamic prompt template, and then formatting that template.  When calling the generate function you can, as a workaround, pass a single character of rag context to ensure the generated prompt includes all the RAG context related system instructions in the prompt, and as result we'll basically get a realistic representation of what the prompt is going to be.  You don't actually need the rag context, you just need a prompt template that assumes there *will be* rag context.

does that make sense?